### PR TITLE
fix: pass name into observe_cloudwatch_logs_subscription

### PR DESCRIPTION
**Pull request by luke-zhu**
## What does this PR do?

This fixes an issue where one could not install multiple terraform-aws-collection modules in the same region.

## Testing

I tested by installing terraform-aws-collection in Thunderdome.

### CODE REVIEW

File: subscription.tf [hunk 1]
```diff
@@ -1,4 +1,6 @@
 module "observe_cloudwatch_logs_subscription" {
+  name = "${var.name}-logs"
```
Comment (jta) [inline subscription.tf:2]: Will this recreate any resources?

Comment (luke-zhu) [reply subscription.tf:2]: yeah, it will re-create everything

Comment (bendrucker) [inline subscription.tf:2]: How does this address the issue? I'd assume the way to apply the module multiple times against the same environment would be to provide a unique `name`. How does adding this suffix resolve the conflict?

Comment (luke-zhu) [reply subscription.tf:2]: we didn't pass in a name before, so observe_cloudwatch_logs_subscription always uses the same default name

Comment (bendrucker) [reply subscription.tf:2]: I see, I misread this as appending `-logs` and not adding a `name`:

https://github.com/observeinc/terraform-aws-cloudwatch-logs-subscription/blob/897508aa88b9c594cc8dbd8275249507dbd8dd5b/variables.tf#L5

Any change that results in replacement of resources should be considered breaking. If you wanted to add this in a non-breaking way, you could add a new variable, e.g., `var.cloudwatch_logs_name`, where `default = null`.

**Review by bendrucker — CHANGES_REQUESTED** [review]

File: subscription.tf (hunk 1 above)
Comment (luke-zhu) [reply subscription.tf:2]: I see, then we can wait until a major version bump to change this. I'll add the variable and a comment then

**Review by bendrucker — APPROVED** [review]
Nice, this at least gives callers a way to ensure uniqueness.

**Pull request merged**